metadata/src/test/java/org/apache/kafka/controller/PartitionChangeBuilderTest.java

```diff
@@ -207,7 +211,7 @@ public void testTriggerLeaderEpochBumpIfNeeded() {
             createFooBuilder()
                 .setTargetIsrWithBrokerStates(
                     AlterPartitionRequest.newIsrToSimpleNewIsrWithBrokerEpochs(Arrays.asList(2, 1, 3, 4)))
-                .setBumpLeaderEpochOnIsrShrink(true),
+                .enableBumpLeaderEpochOnIsrShrink(true),
```

Comment:
Hmm. Shouldn't this test fail? Let's add a comment explaining what this test is doing.

Reply:
This case is an ISR expansion, so we don't expect an epoch bump (I added a comment)

Reply:
Got it. Outside the scope of this PR but it looks like we are missing some validation. This is adding 4 to the ISR but 4 is not a replica for this partition. cc @ahuang98